src/main/java/seedu/duke/model/template/BooleanTemplate.java

```diff
@@ -138,6 +146,31 @@ private String getInputEquation(int index) throws DukeException {
         return equation;
     }
 
+    /**
+     * Recursive function which calculates the output value of the circuit at position passed.
+     *
+     * @param idx int type index of node.
+     * @return int type value output, can be 0 or 1.
+     * @throws DukeException If input values are not set.
+     */
+    public int calculateOutput(int idx) throws DukeException {
```

Comment:
This could be overloaded so that `calculateOutput()` will just give `calculateOutput(0)` when called from `CalculateBooleanCommand`. Then this method could also be `private`.

Reply:
okay noted